### PR TITLE
Feat/manager 13566 Create KMS

### PR DESCRIPTION
**Pull request by vovh**
| Question         | Answer
| ---------------- | ---
| Branch?          |  feat/key-management-service
| Bug fix?         |  no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-13566
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

As a customer, I want to be able to create an KMS to stock all the encryption methods

### CODE REVIEW

File: packages/manager/apps/key-management-service/src/api/GET/apiv2/services.ts [hunk 1]
```diff
@@ -125,3 +125,34 @@ export const sortOKMS = (okms: OKMS[], sorting: ColumnSort): OKMS[] => {
...
+};
+
+export type OKMSCatalog = {
+  plans: OKMSCatalogPlan[];
+};
+
+export const getOrderCatalogOKMS = async (
+  ovhSubsidiary: string,
+): Promise<OKMSCatalog> => {
+  const { data } = await apiClient.v6.get<OKMSCatalog>(
```
Comment (aboungnaseng-ovhcloud) [inline packages/manager/apps/key-management-service/src/api/GET/apiv2/services.ts:154]: why not put this function in an other file ? its a v6 api endpoint, you can check the new folder structure

File: packages/manager/apps/key-management-service/src/pages/create/index.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  ODS_THEME_TYPOGRAPHY_LEVEL,
+} from '@ovhcloud/ods-common-theming';
+import { useTranslation } from 'react-i18next';
+import { ROUTES_URLS } from '@/routes/routes.constants';
+import RegionSelector from '@/components/layout-helpers/Create/RegionSelector';
+import OrderConfirmation from '@/components/layout-helpers/Create/OrderConfirmation';
+
+export default function Create() {
+  const { t } = useTranslation('key-management-service/create');
+  const [orderInitiated, setOrderInitiated] = useState(false);
```
Comment (darsene) [inline packages/manager/apps/key-management-service/src/pages/create/index.tsx:21]: boolean property should start with `is...`

**Review by darsene — APPROVED** [review]

File: packages/manager/apps/key-management-service/src/pages/create/index.tsx (hunk 2 above)
Comment (vovh) [reply packages/manager/apps/key-management-service/src/pages/create/index.tsx]: refactored with is..

File: packages/manager/apps/key-management-service/package.json [hunk 3]
```diff
@@ -25,6 +25,7 @@
     "@ovh-ux/manager-core-api": "*",
     "@ovh-ux/manager-core-utils": "*",
     "@ovh-ux/manager-react-shell-client": "^0.4.0",
+    "@ovh-ux/manager-module-order": "^0.2.0",
```
Comment (chipp972) [inline packages/manager/apps/key-management-service/package.json:28]:
```suggestion
    "@ovh-ux/manager-module-order": "*",
```

File: packages/manager/apps/key-management-service/src/components/layout-helpers/Create/OrderConfirmation.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+};
+
+const OrderConfirmation = ({ region }: OrderConfirmationProps) => {
+  const { t } = useTranslation('key-management-service/create');
+  const [orderLink, setOrderLink] = useState(null);
+  const orderBaseUrl = useOrderURL('express_review_base');
+  const navigate = useNavigate();
+  const generateExpressOrderLink = () => {
+    setOrderLink(
+      `${orderBaseUrl}?products=~(~(productId~'okms~planCode~'okms~duration~'P1M~pricingMode~'default~configuration~(~(label~'region~value~'${region}))))`,
```
Comment (chipp972) [inline packages/manager/apps/key-management-service/src/components/layout-helpers/Create/OrderConfirmation.tsx:38]: Maybe you can add it in the order lib directly
Like I did for the function `getVrackServicesProductSettings`

File: packages/manager/apps/key-management-service/src/components/layout-helpers/Create/OrderConfirmation.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    );
+  };
+
+  useEffect(() => {
+    generateExpressOrderLink();
+  }, []);
```
Comment (chipp972) [inline packages/manager/apps/key-management-service/src/components/layout-helpers/Create/OrderConfirmation.tsx:44]: it doesn't change when the region changes ?
Maybe you don't need a state here

File: packages/manager/apps/key-management-service/src/pages/create/index.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        size={ODS_THEME_TYPOGRAPHY_SIZE._800}
+        color={ODS_THEME_COLOR_INTENT.text}
+      >
+        {t('key_management_service_create_subtitle')}
+      </OsdsText>
+      <OsdsDivider></OsdsDivider>
+      <Notifications />
+      {!isOrderInitiated ? (
+        <RegionSelector
+          orderInitiated={setIsOrderInitiated}
```
Comment (chipp972) [inline packages/manager/apps/key-management-service/src/pages/create/index.tsx:53]:
```suggestion
          setOrderInitiated={() => setIsOrderInitiated(true)}
```

if you don't want to pass `true` to the function inside `RegionSelector`

**Review by chipp972 — APPROVED** [review]

**Pull request merged**